presto-main/src/main/java/com/facebook/presto/operator/scalar/DateTimeFunctions.java

```diff
@@ -556,6 +564,18 @@ public static long diffTimestampWithTimeZone(
         return getTimestampField(unpackChronology(timestampWithTimeZone1), unit).getDifferenceAsLong(unpackMillisUtc(timestampWithTimeZone2), unpackMillisUtc(timestampWithTimeZone1));
     }
 
+    @Description("difference of the given dates in the given unit")
+    @ScalarFunction("date_diff")
+    @LiteralParameters("x")
+    @SqlType(StandardTypes.BIGINT)
+    public static long diffUnixtime(
```

Comment:
I feel like I must be missing something obvious. How does Presto know which of these Java functions to call for the single SQL date_diff fiunction?

Reply:
@elharo These date_diff functions differ in the data type of the parameters. Presto based on that to choose the corresponding function.

Reply:
thanks, I see it now. It's StandardTypes.INTEGER vs StandardTypes.TIME